scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -1290,6 +1290,62 @@
             (catch Exception e
               (log/error e "Unable to decline offers!")))))))
 
+(defn create-mesos-scheduler
+  "Creates the mesos scheduler which offloads processing of status updates to an agent.
+   The agent is used to ensure in-order asynchronous processing of status updates."
```

Comment:
Something like: `(async-in-order (...))`

Also, how does `agent` handle the queue of requests? Is it unbounded? Is there back pressure?